apps/os/backend/agent/mcp/mcp-event-hooks.ts

```diff
@@ -382,6 +396,39 @@ export async function handleMCPConnectRequest(
 
   mcpConnectionCache.managers.set(cacheKey, manager);
 
+  // Ensure mcpConnection row exists for this connection
+  // This upserts based on the partial unique indexes for personal/company modes
+  await db.transaction(async (tx) => {
+    // Check if connection already exists
+    const existingConn = await tx.query.mcpConnection.findFirst({
+      where: and(
+        eq(schema.mcpConnection.estateId, estateId),
+        eq(schema.mcpConnection.serverUrl, serverUrl),
+        mode === "personal"
+          ? and(eq(schema.mcpConnection.mode, "personal"), eq(schema.mcpConnection.userId, userId))
+          : eq(schema.mcpConnection.mode, "company"),
+      ),
+    });
+
+    if (!existingConn) {
+      // Create the connection row
+      await tx.insert(schema.mcpConnection).values({
+        serverUrl,
+        mode,
+        userId: mode === "personal" ? userId : null,
+        estateId,
+        authType: "params", // Will be updated to 'oauth' if OAuth is used later
+        integrationSlug: guaranteedIntegrationSlug,
+      });
+    } else {
+      // Update connected_at timestamp
+      await tx
+        .update(schema.mcpConnection)
+        .set({ connectedAt: new Date(), updatedAt: new Date() })
+        .where(eq(schema.mcpConnection.id, existingConn.id));
+    }
+  });
```

Comment:
in general i think it'd be better if we could use a onConflictDoUpdate type clause, is that possible here? rather than `.findFirst(...)`, `if (!exists) .insert(...) else .update(...)`